py-rattler/src/component.rs

```diff
@@ -0,0 +1,30 @@
+use pyo3::{IntoPy, PyObject, Python};
+use rattler_conda_types::Component;
+
+pub enum PyComponent {
+    String(String),
+    Number(u64),
+    None,
+}
+
+impl IntoPy<PyObject> for PyComponent {
+    fn into_py(self, py: Python) -> PyObject {
+        match self {
+            Self::Number(val) => val.into_py(py),
+            Self::String(val) => val.into_py(py),
+            Self::None => py.None(),
+        }
+    }
+}
+
+impl From<Component> for PyComponent {
+    fn from(value: Component) -> Self {
+        match value {
+            Component::Iden(v) => Self::String(v.to_string()),
+            Component::Numeral(n) => Self::Number(n),
+            Component::Dev => Self::String("dev".to_string()),
+            Component::Post => Self::String("post".to_string()),
+            _ => Self::None,
```

Comment:
```suggestion
            Component::UnderscoreOrDash { .. } => Self::String(String:;from("_")),
```

You can interpret the underscore or dash variant as an `_`. This can happen when you have for instance this version:

```
1.0_
```

The `-` (dash) is normalized to `_` so in this context I think its fine to always use `-`. The only reason there is a difference on the rust side is so that we can reconstruct the version string again.

This allows you to get rid of the `None` variant.